x-pack/plugins/cases/server/services/user_actions/operations/find.ts

```diff
@@ -0,0 +1,219 @@
+
+      const finalFilter = combineFilters([filter, UserActionFinder.buildFilter(types)]);
+
+      const userActions =
+        await this.context.unsecuredSavedObjectsClient.find<CaseUserActionAttributesWithoutConnectorId>(
+          {
+            type: CASE_USER_ACTION_SAVED_OBJECT,
+            hasReference: { type: CASE_SAVED_OBJECT, id: caseId },
+            page,
+            perPage,
+            sortField: 'created_at',
+            sortOrder,
+            filter: finalFilter,
+          }
+        );
+
+      return transformFindResponseToExternalModel(
+        userActions,
+        this.context.persistableStateAttachmentTypeRegistry
+      );
+    } catch (error) {
+      this.context.log.error(`Error finding user actions for case id: ${caseId}: ${error}`);
+      throw error;
+    }
+  }
+
+  private static buildFilter(types: FindOptions['types']) {
+    let typeFilter: KueryNode | undefined;
+
+    for (const type of types) {
+      typeFilter = combineFilters(
+        [typeFilter, UserActionFinder.buildFilterType(type)],
+        NodeBuilderOperators.or
+      );
+    }
+
+    return typeFilter;
+  }
+
+  private static buildFilterType(type: FindTypeField): KueryNode | undefined {
+    switch (type) {
+      case 'action':
+        return UserActionFinder.buildActionFilter();
+      case 'user':
+      case 'alert':
+        return UserActionFinder.buildUserAndAlertFilter(type);
+      case 'attachment':
+        return UserActionFinder.buildAttachmentsFilter();
+      default:
+        return UserActionFinder.buildGenericTypeFilter(type);
+    }
+  }
+
+  private static buildActionFilter(): KueryNode | undefined {
+    const filterForUserActionsExcludingComment = fromKueryExpression(
+      `not ${CASE_USER_ACTION_SAVED_OBJECT}.attributes.type: ${ActionTypes.comment}`
+    );
+
+    return filterForUserActionsExcludingComment;
+  }
+
+  private static buildUserAndAlertFilter(
+    type: typeof FindTypes.alert | typeof FindTypes.user
+  ): KueryNode | undefined {
+    return combineFilters(
+      [
+        buildFilter({
+          filters: [ActionTypes.comment],
+          field: 'type',
+          operator: 'or',
+          type: CASE_USER_ACTION_SAVED_OBJECT,
+        }),
+        buildFilter({
+          filters: [type],
+          field: 'payload.comment.type',
+          operator: 'or',
+          type: CASE_USER_ACTION_SAVED_OBJECT,
+        }),
+      ],
+      NodeBuilderOperators.and
+    );
+  }
+
+  private static buildAttachmentsFilter(): KueryNode | undefined {
+    return combineFilters(
+      [
+        buildFilter({
+          filters: [ActionTypes.comment],
+          field: 'type',
+          operator: 'or',
+          type: CASE_USER_ACTION_SAVED_OBJECT,
+        }),
+        buildFilter({
+          filters: [
+            CommentType.persistableState,
+            CommentType.externalReference,
+            CommentType.actions,
```

Comment:
I think the actions should not be considered an attachment. To me, it is more like an action or a comment. This is how it looks

<img width="2300" alt="Screenshot 2023-01-18 at 3 03 21 PM" src="https://user-images.githubusercontent.com/7871006/213178799-37a78fc9-4c1b-49b3-bfad-d29fc06844bb.png">

Where do you think is the best fit?

Reply:
Sounds good, how about we return them when `types` include `user`?

Reply:
On one hand, the user did an action (isolate the host), but on the other hand, the user typed a comment to explain the action. If we consider our `actions` as system actions meaning actions performed from the case UX, then probably is more appropriate to return it when the `types` includes `user`.